rules/helpers.go

```diff
@@ -82,8 +82,40 @@ func NewVectorAggregation(aggrTypeStr string, vector ast.Node, groupBy clientmod
 	return ast.NewVectorAggregation(aggrType, vector.(ast.VectorNode), groupBy, keepExtraLabels), nil
 }
 
+// vectorMatching combines data used to match samples between vectors.
+type vectorMatching struct {
+	matchCardinality ast.VectorMatchCardinality
+	matchOn          clientmodel.LabelNames
+	includeLabels    clientmodel.LabelNames
+}
+
+// newVectorMatching is a convenience function to create a new vectorMatching.
+func newVectorMatching(card string, matchOn, include clientmodel.LabelNames) (*vectorMatching, error) {
+	var matchCardinalities = map[string]ast.VectorMatchCardinality{
+		"":            ast.MatchOneToOne,
+		"GROUP_LEFT":  ast.MatchManyToOne,
+		"GROUP_RIGHT": ast.MatchOneToMany,
+	}
+	matchCard, ok := matchCardinalities[card]
+	if !ok {
+		return nil, fmt.Errorf("invalid vector match cardinality %q", card)
+	}
+	if matchCard != ast.MatchOneToOne && len(include) == 0 {
+		return nil, fmt.Errorf("group modified vector matching must provide labels")
```

Comment:
Hmm, "group modified"? I guess either just "grouped vector matching" or "group modifier in vector matching"?